### PR TITLE
Add benchmarks for Substring.dropFirst(_:) and dropLast(_:)

### PR DESCRIPTION
**Pull request by natecook1000**
These are self-slicing operations, so they should be fast even for this non-random-access collection.

Hoping to see a difference in this new benchmark from the change in https://github.com/apple/swift/pull/32451.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by natecook1000** [comment]
@swift-ci Please smoke test and merge

File: benchmark/single-source/Substring.swift [hunk 1]
```diff
@@ -126,6 +130,22 @@ public func run_EqualSubstringSubstringGenericEquatable(_ N: Int) {
   }
 }
 
+@inline(never)
+public func run_SubstringDropFirst1(_ N: Int) {
+  let s = quiteLong[...]
+  for _ in 1...N*500 {
```
Comment (xwu) [inline benchmark/single-source/Substring.swift:136]: Maybe 5000 to get a more reliable benchmark?

**Comment by natecook1000** [comment]
@swift-ci Please smoke benchmark

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke benchmark

**Comment by natecook1000** [comment]
@swift-ci Please smoke benchmark

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Pull request merged**